### DISCUSSION TITLE
Question about crystal geometry

### DISCUSSION THREAD

**Discussion by bjzhou96** [mtex-toolbox/mtex#1391] [Ask Anything] [answered]
Hello everyone

I am confused by some questions on crystal geometry. I conducted EBSD analysis on olivine grains in a thin section in two laboratories (Lab 1 by mapping and Lab 2 by pointting). However, I got two different pole figures. Then I check the .ctf files and find the possible reason. In Lab 1, the olivine grains were indexed as 'Olivine' phase, with a crystal symmetry of  ('mmm', [10 6 4.7], 'mineral', 'Olivine'); while in Lab 2, the olivine grains were indexed as 'Forsterite' phase with a crystal sysmetry of ('mmm', [4.8 10 6], 'mineral', 'Forsterite'). Here is the problem, the order of three crystal axes in the databases of two labs are different, which may result in the different patterns of pole figures. I am wondering which one I should choose and why this happens.

I appreciate it very much if you can help me out. Thank you in advance.

Best regards,
Baojun Zhou

**Comment by filippeof** [comment]
Olivine can be described in two different space groups: _Pnma_ where a>b>c (your olivine) and _Pbnm_ where b>c>a (your forsterite). In principle, it shouldn't mattter which one you use to index your patterns. Is it possible that you have different reference frames in the different labs? (see here: https://mtex-toolbox.github.io/EBSDReferenceFrame.html)
Maybe it would help if you include your script and some data
Best,
Filippe

**Comment by bjzhou96** [comment]
Dear Filippe

Thank you for your reply! The reference frames in two labs are supposed to be same, because the CPO patterns of other mineral (e.g. amphibole) in another thin section obtained in two labs are same. If it is not about olivine crystal geometry, I totally have no idea what is going wrong. Thank you for helping me!

Here is the script for two data

For data by mapping:
```
%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('mmm', [10 6 4.7], 'mineral', 'Olivine', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('12/m1', [9.9 18 5.4], [90,104.81,90]*degree, 'X||a*', 'Y||b*', 'Z||c', 'mineral', 'Amphibole', 'color', [0.56 0.74 0.56]),...
  crystalSymmetry('mmm', [19 9 5.3], 'mineral', 'Orthopyroxene', 'color', [0.85 0.65 0.13]),...
  crystalSymmetry('12/m1', [9.8 9 5.3], [90,106.03,90]*degree, 'X||a*', 'Y||b*', 'Z||c', 'mineral', 'Clinopyroxene', 'color', [0.94 0.5 0.5]),...
  crystalSymmetry('m-3m', [8.4 8.4 8.4], 'mineral', 'Magnetite', 'color', [0 0 0.55]),...
  crystalSymmetry('1', [10 6 9.5], [9.88,9.88,9.88]*degree, 'X||a*', 'Z||c', 'mineral', 'Olivines', 'color', [0 0.39 0]),...
  crystalSymmetry('m-3m', [8.2 8.2 8.2], 'mineral', 'Spinel - (Mg,Fe)2SiO4', 'color', [0.55 0 0])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'D:\西藏\橄榄岩变形\EBSD\YJ2018_2';

% which files to be imported
fname = [pname '\YJ2018-2reduction.cpr'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','crc',...
  'convertSpatial2EulerReferenceFrame');
[grains,ebsd.grainId] = calcGrains(ebsd);
ebsd(grains(grains.grainSize<10)) = [];
[grains,ebsd.grainId] = calcGrains(ebsd,'angel','10*degree');
OnePointPerGrains = grains('Olivine').meanOrientation;
odf = calcDensity(OnePointPerGrains,'halfwidth',10*degree);
% the fixed crystal directions
PF1AMP1 = Miller(1,1,0,ebsd('Olivine').CS,'hkl');
PF1AMP2 = Miller(1,0,0,ebsd('Olivine').CS,'hkl');
PF1AMP3 = Miller(0,1,0,ebsd('Olivine').CS,'hkl');
PF1AMP4 = Miller(0,0,1,ebsd('Olivine').CS,'uvw');
PF1AMP5 = Miller(1,1,0,ebsd('Olivine').CS,'uvw');
PF1AMP6 = Miller(1,-1,0,ebsd('Olivine').CS,'uvw');
h = {PF1AMP1,PF1AMP2,PF1AMP3,PF1AMP4,PF1AMP5,PF1AMP6};
% Plot figures colorful
plotPDF(odf,h,'lower','resolution',2*degree,'contourf');
CLim(gcm,'equal');
mtexColorbar
%% 三-2、Fabric Strength - Jindex and Mindex
Jindex= textureindex(odf);
Mindex= calcMIndex(odf);

% show the value of fabric strengh
disp(' ')
disp(['Jindex = ',num2str(Jindex,3)])
disp(['Mindex = ',num2str(Mindex,3)])
disp(' ')

For data by pointing

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('-3m1', [4.9 4.9 5.5], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Quartz-new', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('-3m1', [5 5 17], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Calcite', 'color', [0.56 0.74 0.56]),...
  crystalSymmetry('12/m1', [8.6 13 7.2], [90,116,90]*degree, 'X||a*', 'Y||b*', 'Z||c', 'mineral', 'Orthoclase', 'color', [0.85 0.65 0.13]),...
  crystalSymmetry('-1', [8.2 13 7.1], [94.06,116.5,88.59]*degree, 'X||a*', 'Z||c', 'mineral', 'Oligoclase', 'color', [0.94 0.5 0.5]),...
  crystalSymmetry('12/m1', [5.3 9.3 10], [90,100.23,90]*degree, 'X||a*', 'Y||b*', 'Z||c', 'mineral', 'Biotite', 'color', [0 0 0.55]),...
  crystalSymmetry('m-3m', [5.4 5.4 5.4], 'mineral', 'Silicon', 'color', [0 0.39 0]),...
  crystalSymmetry('12/m1', [9.9 18 5.3], [90,104.86,90]*degree, 'X||a*', 'Y||b*', 'Z||c', 'mineral', 'Amphibole', 'color', [0.55 0 0]),...
  crystalSymmetry('12/m1', [8.5 13 7.2], [90,115.9,90]*degree, 'X||a*', 'Y||b*', 'Z||c', 'mineral', 'Feldspar', 'color', [0.58 0 0.83]),...
  crystalSymmetry('mmm', [4.8 10 6], 'mineral', 'Forsterite', 'color', [0.5 0.5 0.5]),...
  crystalSymmetry('4/m', [12 12 12], 'mineral', 'Garnet', 'color', [0.85 0.75 0.85]),...
  crystalSymmetry('m-3m', [8.4 8.4 8.4], 'mineral', 'Magnetite', 'color', [0.5 1 0.83]),...
  crystalSymmetry('mmm', [19 9 5.3], 'mineral', 'Orthopyroxene', 'color', [0.24 0.7 0.44]),...
  crystalSymmetry('12/m1', [17 17 7.3], [90,116.44,90]*degree, 'X||a*', 'Y||b*', 'Z||c', 'mineral', 'Clinoptilolite', 'color', [1 0.63 0.48]),...
  crystalSymmetry('12/m1', [9.7 8.9 5.3], [90,106.34,90]*degree, 'X||a*', 'Y||b*', 'Z||c', 'mineral', 'Clinopyroxene diopsid', 'color', [0.27 0.51 0.71])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'D:\西藏\橄榄岩变形\EBSD\YJ2018_2';

% which files to be imported
fname = [pname '\zong.ctf'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ctf',...
  'convertSpatial2EulerReferenceFrame');

[grains,ebsd.grainId] = calcGrains(ebsd);
OnePointPerGrains = grains('Forsterite').meanOrientation;
odf = calcDensity(OnePointPerGrains,'halfwidth',10*degree);
% the fixed crystal directions
PF1AMP1 = Miller(1,1,0,ebsd('Forsterite').CS,'hkl');
PF1AMP2 = Miller(1,0,0,ebsd('Forsterite').CS,'hkl');
PF1AMP3 = Miller(0,1,0,ebsd('Forsterite').CS,'hkl');
PF1AMP4 = Miller(0,0,1,ebsd('Forsterite').CS,'uvw');
PF1AMP5 = Miller(1,1,0,ebsd('Forsterite').CS,'uvw');
PF1AMP6 = Miller(1,-1,0,ebsd('Forsterite').CS,'uvw');
h = {PF1AMP1,PF1AMP2,PF1AMP3,PF1AMP4,PF1AMP5,PF1AMP6};
% Plot figures colorful
plotPDF(odf,h,'lower','resolution',2*degree,'contourf');
CLim(gcm,'equal');
mtexColorbar
%% 三-2、Fabric Strength - Jindex and Mindex
Jindex= textureindex(odf);
Mindex= calcMIndex(odf);

% show the value of fabric strengh
disp(' ')
disp(['Jindex = ',num2str(Jindex,3)])
disp(['Mindex = ',num2str(Mindex,3)])
disp(' ')
```
[data.zip](https://github.com/mtex-toolbox/mtex/files/8388602/data.zip)

**Comment by marcoalopez** [comment]
Hi,

just a quick tip not directly related to your question. When copying code on github take advantage of the possibilities of the markdown language and include your code in markdown "code fences" as follows

\```matlab
_code here_
\```

so that the code is formatted and highlighted appropriately and makes it easier to view. For example:

```matlab
% crystal symmetry
CS = {...
'notIndexed',...
crystalSymmetry('-3m1', [4.9 4.9 5.5], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Quartz-new', 'color', [0.53 0.81 0.98]),...
crystalSymmetry('-3m1', [5 5 17], 'X||a*', 'Y||b', 'Z||c*', 'mineral', 'Calcite', 'color', [0.56 0.74 0.56]),...
crystalSymmetry('12/m1', [8.6 13 7.2], [90,116,90]degree, 'X||a', 'Y||b*', 'Z||c', 'mineral', 'Orthoclase', 'color', [0.85 0.65 0.13]),...
crystalSymmetry('-1', [8.2 13 7.1], [94.06,116.5,88.59]degree, 'X||a', 'Z||c', 'mineral', 'Oligoclase', 'color', [0.94 0.5 0.5]),...
crystalSymmetry('12/m1', [5.3 9.3 10], [90,100.23,90]degree, 'X||a', 'Y||b*', 'Z||c', 'mineral', 'Biotite', 'color', [0 0 0.55]),...
crystalSymmetry('m-3m', [5.4 5.4 5.4], 'mineral', 'Silicon', 'color', [0 0.39 0]),...
crystalSymmetry('12/m1', [9.9 18 5.3], [90,104.86,90]degree, 'X||a', 'Y||b*', 'Z||c', 'mineral', 'Amphibole', 'color', [0.55 0 0]),...
crystalSymmetry('12/m1', [8.5 13 7.2], [90,115.9,90]degree, 'X||a', 'Y||b*', 'Z||c', 'mineral', 'Feldspar', 'color', [0.58 0 0.83]),...
crystalSymmetry('mmm', [4.8 10 6], 'mineral', 'Forsterite', 'color', [0.5 0.5 0.5]),...
crystalSymmetry('4/m', [12 12 12], 'mineral', 'Garnet', 'color', [0.85 0.75 0.85]),...
crystalSymmetry('m-3m', [8.4 8.4 8.4], 'mineral', 'Magnetite', 'color', [0.5 1 0.83]),...
crystalSymmetry('mmm', [19 9 5.3], 'mineral', 'Orthopyroxene', 'color', [0.24 0.7 0.44]),...
crystalSymmetry('12/m1', [17 17 7.3], [90,116.44,90]degree, 'X||a', 'Y||b*', 'Z||c', 'mineral', 'Clinoptilolite', 'color', [1 0.63 0.48]),...
crystalSymmetry('12/m1', [9.7 8.9 5.3], [90,106.34,90]degree, 'X||a', 'Y||b*', 'Z||c', 'mineral', 'Clinopyroxene diopsid', 'color', [0.27 0.51 0.71])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');
```

cheers.
Marco

**Reply by bjzhou96** [reply]
Dear Marco

Thank you for your advice! I am sorry for my inappropriate expressions.

Best regards,
Baojun Zhou

**Comment by zmichels** [comment]
It actually does matter (as indicated by the original post), and I’ve seen this before… because the axes lengths are mapped differently than the actual mineral — The pole figures all end up with axes looking swapped, and they make no sense or cause much confusion. The ‘Forsterite’ match unit has always given me the correct results, but a handful of ‘olivine’ match units lead to incorrect indexing for the olivine I have analyzed (because the axes lengths are different than the particular olivine I was analyzing). Presumably it is appropriate for a different flavor of olivine, but I have yet to encounter it, myself. If the data were collected with a mismatched match-unit, thankfully MTEX can be used to compute correct orientations from the existing ones and then assign the newly computed orientations to the ebsd variable — I have had to do this for a couple colleagues to help salvage maps that were collected with an inappropriate match unit. 

Zach

**Comment by kilir** [comment]
Hi,
in order to import the one with the funny ordering of a,b,c into mtex, you should add the proper alignment to the CS definition. E.g.
```
 crystalSymmetry('mmm', [10 6 4.7], 'mineral', 'Olivine', 'color', [0.53 0.81 0.98], ` 'X||b', 'Y||c', 'Z||a'`),...
```
assuming that your indexing software knew how to deal with the material file to begin with.

Totally unrelated, I'm happy to see that there isn't quartz and forsterite indexed together.

Cheers,
Rüdiger

**Comment by bjzhou96** [comment]
Hi,

Thank you very much for your help! The problem has been solved aoocrding to your advice. 

Best regard,
Baojun Zhou

**Comment by filippeof** [comment, marked as answer]
@bjzhou96, also be aware that your cpr/crc files have two _olivine_ crystal symmetries and one of them is triclinic (1). I guess you manually renamed it to _olivines_ but it is probably better to rename it to something else as mtex might understand _olivine_ as an abbreviation to _olivines_.
@zmichels For EBSD indexing, the choice between Pbnm or Pnma seems to be more about (personal) taste than (olivine) flavour. Considering that the space group is reported, it shouldn't matter, as long as a,b,c in Pnma is compared with b,c,a in Pbnm. I agree that confusion can be created when abc in Pnma is compared with abc in Pbnm. 
As Pbnm is more commonly used for natural olivine structure (e.g. Birle et al. 1968), we should probably stick to it or rotate the data indexed in Pnma as @kilir mentioned. And equally importantly, we should always report which space group we use :)

Birle, J. D., G. V. Gibbs, P. B. Moore, and J. V. Smith. ‘Crystal Structures of Natural Olivines’. American Mineralogist 53, no. 5–6 (1 June 1968): 807–24.

Best,
Filippe

**Reply by bjzhou96** [reply]
Dear Filippe

Thank you very much for your advice and detailed explanation! It is very kind of you to share this useful article with me! I do benefit a lot.

Best regard,
Baojun Zhou
